### PR TITLE
Adding SecurityType to ProtoBuf

### PR DESCRIPTION
**Pull request by SethBe**
Adding Security Type to protobufs. This is a foundational element used by all components and will be built upon

### CODE REVIEW

File: rpc/cloudagent/compute/virtualmachine/moc_cloudagent_virtualmachine.proto [hunk 1]
```diff
@@ -96,9 +96,16 @@ message UefiSettings {
 	bool secureBootEnabled = 1;
 }
 
+enum {
```
Comment (SethBe) [inline rpc/cloudagent/compute/virtualmachine/moc_cloudagent_virtualmachine.proto:99]: bug, accidently ctrl-z will fix

**Review by danmamsft — APPROVED** [review]

File: rpc/cloudagent/compute/virtualmachine/moc_cloudagent_virtualmachine.proto [hunk 2]
```diff
@@ -96,9 +96,16 @@ message UefiSettings {
 	bool secureBootEnabled = 1;
 }
 
+enum SecurityType {
```
Comment (raghavendra-nataraj) [inline rpc/cloudagent/compute/virtualmachine/moc_cloudagent_virtualmachine.proto:99]: Can we add this to https://github.com/microsoft/moc/blob/main/rpc/common/moc_common_computecommon.proto. This way we can avoid duplicate definitions

Comment (SethBe) [reply rpc/cloudagent/compute/virtualmachine/moc_cloudagent_virtualmachine.proto:99]: Makes sense ill give it a go

Comment (SethBe) [reply rpc/cloudagent/compute/virtualmachine/moc_cloudagent_virtualmachine.proto]: Note this triggers changes in other repros, i will update those Pr's after this completes

File: rpc/common/moc_common_computecommon.proto [hunk 3]
```diff
@@ -154,3 +154,9 @@ enum Architecture {
 extend google.protobuf.FieldOptions {
 	bool sensitivecompute = 50002;
 }
+
+enum SecurityType {
+        NOTCONFIGURED = 0;
+        TRUSTEDLAUNCH = 1;
+        CONFIDENTIALVM = 2;
+}
```
Comment (SethBe) [inline rpc/common/moc_common_computecommon.proto:162]: Nit space

File: rpc/cloudagent/compute/virtualmachine/moc_cloudagent_virtualmachine.proto (hunk 2 above)
Comment (SethBe) [reply rpc/cloudagent/compute/virtualmachine/moc_cloudagent_virtualmachine.proto]: Pr's are updated

**Review by raghavendra-nataraj — APPROVED** [review]

**Pull request merged**